### PR TITLE
[SW-173358] add first token prints

### PR DESCRIPTION
**Pull request by shakkedw**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by dvarshney-habana — COMMENTED** [review]
@shakkedw  
I cannot understand the purpose of the prints you added in optimum/habana/transformers/generation/utils.py

Can you please share a sample log with your changes. (enough just for greedy case)

File: examples/text-generation/run_generation.py [hunk 1]
```diff
@@ -289,6 +289,8 @@ def main():
         def generate(size=None, reduce_recompile=False):
             """Generates sequences from the input sentences and returns them."""
 
+            t0 = time.perf_counter()
+            print(f"Step4+ starting time is {t0*1000}", flush=True)
```
Comment (puneeshkhanna) [inline examples/text-generation/run_generation.py:293]: Is there a need to print starting time ?

File: examples/text-generation/run_generation.py [hunk 2]
```diff
@@ -317,7 +319,10 @@ def generate(size=None, reduce_recompile=False):
                 profiling_steps=args.profiling_steps,
                 profiling_warmup_steps=args.profiling_warmup_steps,
             ).cpu()
-            return tokenizer.batch_decode(outputs, skip_special_tokens=True)
+            x = tokenizer.batch_decode(outputs, skip_special_tokens=True)
```
Comment (puneeshkhanna) [inline examples/text-generation/run_generation.py:322]: Lets rename outputs to output_tokens and x to outputs for better convention and understandibility.

**Comment by shakkedw** [comment]
@dvarshney-habana, @puneeshkhanna 
the purpose for those prints is to measure first token latency. coming from the ticket - https://jira.habana-labs.com/browse/SW-169485
you can see the prints in the logs:
/software/users/sweinberger/infer/models/file-llama-7b.txt
/software/users/sweinberger/infer/models/file-llama-70b.txt
/software/users/sweinberger/infer/models/file-falcon-40b.txt
/software/users/sweinberger/infer/models/file-gpt-j-6b.txt

**Comment by puneeshkhanna** [comment]
Changes look good to me.
@shakkedw  - make style done ?

**Review by dvarshney-habana — CHANGES_REQUESTED** [review]
we need to only print the latency for first token by taking a timestamp before and after the generation
Current prints are not measuring that.
Also , print in examples/text-generation/run_generation.py prints absolute at start of first token. Also, pls print all time units (preferred in ms)

**Review by dvarshney-habana — APPROVED** [review]

**Pull request merged**